aptos-move/move-examples/token_objects/guild/sources/guild.move

```diff
@@ -0,0 +1,309 @@
+/// This module implements the guild token (non-fungible token) including the
+/// functions create the collection and the guild tokens.
+module token_objects::guild {
+    use std::option;
+    use std::signer;
+    use std::string::{Self, String};
+    use std::vector;
+    use aptos_framework::object::{Self, Object};
+    use aptos_token_objects::collection;
+    use aptos_token_objects::token;
+
+    /// The token does not exist
+    const ETOKEN_DOES_NOT_EXIST: u64 = 1;
+    /// The provided signer is not the admin
+    const ENOT_ADMIN: u64 = 2;
+    /// The provided signer is not the owner
+    const ENOT_OWNER: u64 = 3;
+    /// The provided signer is not a guild master
+    const ENOT_GUILD_MASTER: u64 = 4;
+
+    /// The guild token collection name
+    const GUILD_COLLECTION_NAME: vector<u8> = b"Guild Collection Name";
+    /// The guild collection description
+    const GUILD_COLLECTION_DESCRIPTION: vector<u8> = b"Guild Collection Description";
+    /// The guild collection URI
+    const GUILD_COLLECTION_URI: vector<u8> = b"https://guild.collection.uri";
+
+    /// Whitelisted guild masters
+    struct Config has key {
+        whitelist: vector<address>,
+    }
+
+    /// Manager object refs
+    struct Manager has key {
+        /// The extend_ref of the manager object to get its signer
+        extend_ref: object::ExtendRef,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Guild token
+    struct GuildToken has key {
+        /// Used to get the signer of the token
+        extend_ref: object::ExtendRef,
+        /// Member collection name
+        member_collection: String,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Member token
+    struct MemberToken has key {
+        /// Belonging guild
+        guild: Object<GuildToken>,
+        /// Used to burn
+        burn_ref: token::BurnRef,
+    }
+
+    /// Initializes the module, creating the manager object, the guild token collection and the whitelist.
+    fun init_module(sender: &signer) acquires Manager {
+        // Create the manager object to manage the guild collection.
+        create_manager(sender);
+        // Create a collection for guild tokens.
+        create_guild_collection(&manager_signer());
+        // Create a whitelist
+        move_to(sender, Config { whitelist: vector::empty() });
+    }
+
+    #[view]
+    /// Returns the guild token address by name
+    public fun guild_token_address(guild_token_name: String): address acquires Manager {
```

Comment:
What is the intended use case for this?

Reply:
This is for a user (guild master) to locate their guild token because `mint_member` requires a guild token as an argument.